src/api/app/services/file_service/uploader.rb

```diff
@@ -0,0 +1,131 @@
+module FileService
+  class Uploader
+    attr_reader :errors, :added_files
+
+    def initialize(package, files, empty_files, file_urls, comment)
+      @commit_filelist = []
+      @errors = []
+      @added_files = []
+
+      @package = package
+      @files = files
+      @empty_files = empty_files
+      @file_urls = file_urls
+      @comment = comment
+    end
+
+    def call
+      fetch_existing_files
+
+      begin
+        empty_files if @empty_files.present?
+        http_upload if @files.present?
```

Comment:
I would rename `empty_files`, `http_upload` and `remote_files` appending them a `!` to signal they are not side-effect free.